.github/workflows/_build-test-node-and-runtime.yml

```diff
@@ -0,0 +1,98 @@
+---
+#  This workflow builds test version of aleph-node and aleph-runtime, ie with some
+#  features enabled. Those binaries MUST NOT be used in the production.
+name: Build test aleph-node and test aleph-runtime
+on:
+  workflow_call:
+
+jobs:
+  main:
+    name: Build binary artifacts
+    runs-on: self-hosted
+    env:
+      RUST_BACKTRACE: full
+      RUSTC_WRAPPER: sccache
+      SECRETS_AWS_MAINNET_ACCESS_KEY_ID: ${{ secrets.AWS_MAINNET_ACCESS_KEY_ID }}
+      SECRETS_AWS_MAINNET_SECRET_ACCESS_KEY: ${{ secrets.AWS_MAINNET_SECRET_ACCESS_KEY }}
+      SECRETS_AWS_DEVNET_ACCESS_KEY_ID: ${{ secrets.AWS_DEVNET_ACCESS_KEY_ID }}
+      SECRETS_AWS_DEVNET_SECRET_ACCESS_KEY: ${{ secrets.AWS_DEVNET_SECRET_ACCESS_KEY }}
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Get branch name and commit SHA
+        id: get_branch
+        uses: ./.github/actions/get-branch
```

Comment:
Shouldn't we use freshly refactored version of this action from https://github.com/Cardinal-Cryptography/aleph-node/pull/1012?
It applies not only to this workflow.

Reply:
Yes, but the other PR is nor merged to `main` yet, so I cannot use it here. I'll most likely merge that PR and rebase this one

Reply:
done